op-e2e/custom_gas_token_test.go

```diff
@@ -0,0 +1,456 @@
+package op_e2e
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum-optimism/optimism/op-bindings/bindings"
+	"github.com/ethereum-optimism/optimism/op-bindings/predeploys"
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils"
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils/receipts"
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils/wait"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	"github.com/ethereum-optimism/optimism/op-service/testlog"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCustomGasToken(t *testing.T) {
+	InitParallel(t, SkipOnFPAC) // Custom Gas Token feature is not yet compatible with FPAC
+
+	cfg := DefaultSystemConfig(t)
+	cfg.DeployConfig.FinalizationPeriodSeconds = 2 // 2s finalization period
```

Comment:
i don't think modifying the finalization period does anything as its a contracts thing and the contract has already been deployed. the deploy config is a leaky abstraction in practice

Reply:
Gotcha. I had copy pasted this from https://github.com/ethereum-optimism/optimism/blob/ac0929bf84230e485b1e2c4264b2f52fecb9654e/op-e2e/system_test.go#L1061-L1062 . It doesn't seem to affect the duration of the test, so I removed it.

Reply:
Gotcha. I had copy pasted this from https://github.com/ethereum-optimism/optimism/blob/ac0929bf84230e485b1e2c4264b2f52fecb9654e/op-e2e/system_test.go#L1061-L1062 . It doesn't seem to affect the duration of the test, so I removed it here [701aabd](https://github.com/ethereum-optimism/optimism/pull/10440/commits/701aabd333b1d421a05f7e08e871e0ab53a1664b).